### PR TITLE
Update fmt to 6.2.0

### PR DESCRIPTION
**Pull request by peterbell10**
Isolates fmt changes from #4708 with some additions

- Fixed the `Vector3` formatter
- Use `format_args` like versions everywhere
- Use `fmt::memory_buffer` to avoid some string copying

Works with C++11, might need Appveyor update though.

### CODE REVIEW

File: src/OSSupport/File.h [hunk 1]
```diff
@@ -163,8 +163,12 @@ class cFile
 	/** Returns the list of all items in the specified folder (files, folders, nix pipes, whatever's there). */
 	static AStringVector GetFolderContents(const AString & a_Folder);  // Exported in ManualBindings.cpp
 
-	int Printf(const char * a_Fmt, fmt::ArgList);
-	FMT_VARIADIC(int, Printf, const char *)
+	int vPrintf(const char * a_Fmt, fmt::printf_args);
+	template <typename... Args>
+	int Printf(const char * a_Fmt, const Args & ... a_Args)
+	{
+		return vPrintf(a_Fmt, fmt::make_printf_args(a_Args...));
```
Comment (tigerw) [inline src/OSSupport/File.h:170]: mhm, you might be surprised, this actually increased my binary size by 0.1

**Comment by tigerw** [comment]
oi, no competition

**Comment by tigerw** [comment]
But in all seriousness the other PR is already split up, I propose merging your fancy additions here into that.

**Comment by peterbell10** [comment]
>oi, no competition

Just thought it made sense I take over for fmt since it's me that added it in the first place. This PR should also support the current compiler set so is really independent of the C++17 stuff.

File: Tools/GrownBiomeGenVisualiser/Globals.h [hunk 2]
```diff
@@ -191,7 +191,23 @@ typedef unsigned char Byte;
...
+	#pragma clang diagnostic ignored "-Wc++2a-compat"
+#endif
+
 #include "fmt/format.h"
+#include "fmt/printf.h"
+
+#ifdef __clang__
+	#pragma clang diagnostic pop
+#endif
+
```
Comment (madmaxoft) [inline Tools/GrownBiomeGenVisualiser/Globals.h:210]: Wouldn't it be easier to put all this into one separate include file? `#include "fmt.h"`

File: src/CommandOutput.h [hunk 3]
```diff
@@ -16,8 +16,13 @@ class cCommandOutputCallback
 	virtual ~cCommandOutputCallback() {}  // Force a virtual destructor in subclasses
 
 	/** Syntax sugar function, calls Out() with Printf()-ed parameters; appends a newline" */
-	void Out(const char * a_Fmt, fmt::ArgList);
-	FMT_VARIADIC(void, Out, const char *)
+	void vOut(const char * a_Fmt, fmt::printf_args);
```
Comment (madmaxoft) [inline src/CommandOutput.h:19]: The doxy-comment now doesn't fit

File: src/OSSupport/File.h (hunk 1 above)
Comment (peterbell10) [reply src/OSSupport/File.h:170]: I tried replacing the main `Printf` override in `StringUtils.h` with a templated version and my results were:
clang - 0.1 MB increase
gcc- 0.01 MB increase

Not earth shattering but still worse than before. What compiler are you testing on?

Comment (tigerw) [reply src/OSSupport/File.h:170]: Oh not in StringUtils, just in this one, File.h, Visual C++.

Anyway consistency dictates this is fine.

**Comment by tigerw** [comment]
yeah I'll rebase 4708 off this

**Pull request merged**